chain/rust/src/genesis/network_info.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{
+    byron::ProtocolMagic,
+    plutus::CostModels
+};
+use cml_core::Int;
+
+#[derive(Debug, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct NetworkInfo {
+    network_id: u8,
+    protocol_magic: ProtocolMagic,
+}
+impl NetworkInfo {
+    pub fn new(network_id: u8, protocol_magic: ProtocolMagic) -> Self {
+        Self {
+            network_id,
+            protocol_magic,
+        }
+    }
+    pub fn network_id(&self) -> u8 {
+        self.network_id
+    }
+    pub fn protocol_magic(&self) -> ProtocolMagic {
+        self.protocol_magic
+    }
+
+    pub fn testnet() -> NetworkInfo {
```

Comment:
in any case protocol magics for preview and preprod are different. as we are creating an sdk I think we should support all testnets

Reply:
Are they the same though? In this struct, used for genesis/byron, it's 32-bits and it can definitely use all of those. Elsewhere we have 4-bits of data in addresses, referred to as "network id" so I'm not sure if this is completely unrelated. I'm completely ignorant about protocol magic after byron though so I thought it was replaced with just the network id. The binary spec makes no reference to protocol magics, only network id.

What are the protocol magics for current networks?

Reply:
Ah, is this about CIP34? We could implement the stuff in that registry then I guess. It's not directly part of the on-chain data (post-byron) so we should document what it's referring to I guess. If we do that it might make more sense to move it away from the genesis module then.